yb-voyager/cmd/analyzeSchema.go

```diff
@@ -419,6 +413,138 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
+		location:98  keys:{string:{sval:"id"}} keys:{string:{sval:"country_code"}}}}  partspec:{strategy:PARTITION_STRATEGY_RANGE
+		part_params:{partition_elem:{name:"country_code" location:150}}  part_params:{partition_elem:{name:"record_type"  ...
+	*/
+	if createTableNode.CreateStmt.GetPartspec() == nil {
+		//If not partition table then no need to proceed
+		return
+	}
+
+	if primaryConsInAlter[fullyQualifiedName] != nil {
+		//reporting the ALTER TABLE ADD PK on partition table here in case the order is different if ALTER is before the CREATE
+		alterTableSqlInfo := primaryConsInAlter[fullyQualifiedName]
+		reportCase(alterTableSqlInfo.fileName, ADDING_PK_TO_PARTITIONED_TABLE_ISSUE_REASON,
+			"https://github.com/yugabyte/yugabyte-db/issues/10074", "", "TABLE", fullyQualifiedName, alterTableSqlInfo.formattedStmt, MIGRATION_CAVEATS, ADDING_PK_TO_PARTITIONED_TABLE_DOC_LINK)
+	}
+
+	partitionTablesMap[fullyQualifiedName] = true // marking the partition tables in the map
+
+	var primaryKeyColumns, partitionColumns, uniqueKeyColumns []string
+
+	for _, column := range columns {
+		if column.GetColumnDef() != nil { //In case PRIMARY KEY constraint is added with column definition
+			constraints := column.GetColumnDef().Constraints
+			for _, constraint := range constraints {
+				if constraint.GetConstraint().Contype == pg_query.ConstrType_CONSTR_PRIMARY {
+					primaryKeyColumns = []string{column.GetColumnDef().Colname}
+				}
+				if constraint.GetConstraint().Contype == pg_query.ConstrType_CONSTR_UNIQUE {
+					uniqueKeyColumns = append(uniqueKeyColumns, column.GetColumnDef().Colname)
+				}
+			}
+		} else if column.GetConstraint() != nil {
+			//In case CREATE DDL has PRIMARY KEY(column_name) - it will be included in columns but won't have columnDef as its a constraint
+			for _, key := range column.GetConstraint().GetKeys() {
+				if column.GetConstraint().Contype == pg_query.ConstrType_CONSTR_PRIMARY {
+					primaryKeyColumns = append(primaryKeyColumns, key.GetString_().Sval)
+				} else if column.GetConstraint().Contype == pg_query.ConstrType_CONSTR_UNIQUE {
+					uniqueKeyColumns = append(uniqueKeyColumns, key.GetString_().Sval)
+				}
+			}
+		}
+	}
+
+	partitionElements := createTableNode.CreateStmt.GetPartspec().GetPartParams()
+
+	for _, partElem := range partitionElements {
+		if partElem.GetPartitionElem().GetExpr() != nil {
+			//Expression partitions
+			fmt.Printf("expression par on %s, %v", fullyQualifiedName, primaryKeyColumns)
+			if len(primaryKeyColumns) > 0 || len(uniqueKeyColumns) > 0 {
+				summaryMap["TABLE"].invalidCount[fullyQualifiedName] = true
+				reportCase(fpath, "Issue with Partition using Expression on a table which cannot contain Primary Key / Unique Key on any column",
+					"https://github.com/yugabyte/yb-voyager/issues/698", "Remove the Constriant from the table definition", "TABLE", fullyQualifiedName, sqlStmtInfo.formattedStmt, UNSUPPORTED_FEATURES, EXPRESSION_PARTIITON_DOC_LINK)
+				return
```

Comment:
Is it okay to return here? Should we break instead? 
THere could be other issues as well, right (for example, the ones being detected below...) ?
It would be better to  NOT stop when we find an issue.

Reply:
Yeah agreed, we can report other as well so that user can understand all the issues and modify the DDL accordingly.